src/saturn_engine/utils/asyncutils.py

```diff
@@ -305,7 +314,7 @@ async def wrapper() -> None:
                     future.set_result(value)
                 except BaseException as e:
                     future.set_exception(e)
-                    del instance.__dict__[self._name]
+                    instance.__dict__.pop(self._name, None)
```

Comment:
While working on this I also happen to fix a few bug in the `cached_property` where many task waiting for the same property that fail will raise an error here for all the task execpt the first.